### PR TITLE
Game and Servlet java code

### PR DESCRIPTION
**Pull request by brettallenyo**
Code that allows game to function, changed from just displaying random quotes.

Transitions between two states:
0: User sees the first half of a quote and can enter the rest of it in a text box
1: It either shows "Correct" or "False" and then shows the correct answer, either way it has a button that allows the user to go to the next quote

The Java code uses a HashMap and a current variable to maintain the quotes and the current quote in question

The Servlet code uses the state variable as described above to return the correct JSON

The Javascript code uses the JSON to find out the state and display the correct thing

I separated the two states into two different HTML files

https://screenshot.googleplex.com/pFMAX1HtZWL

https://screenshot.googleplex.com/0idTwifHQUL

Apologies for the massive commit

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  /** Picks a new prompt at random to assign to the current prompt */
+  public void newChoice() {
+    int rand = (int) (Math.random() * values.size());
+    int index = 0;
+    for (String i : values.keySet()) {
+      if (index == rand) {
+        current = i;
+        break;
+      }
+      index++;
```
Comment (luli0521) [inline portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java:88]: Is this randomization the only reason you use a HashMap? If so, you can simply write using a List:
       
 return list.get(new Random().nextInt(list.size()));

File: portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+/**
+ * Class representing the Office game, where players are instructed to finish a prompted quote.
+ *
+ */
+public class OfficeGame {
+  /** Map of values from prompts to answers */
+  private final HashMap<String, String> values = new HashMap<String, String>();
+
+  /** The current prompt to answer */
+  private String current = "I'm Prison ";
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java:30]: currentPrompt

File: portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    values.put("And the worst thing about prison ", "was the dementors");
+    values.put("Identity theft is not a joke, Jim! ", "Millions of families suffer every year");
+    values.put("Bonfire, James Bond-fire. ", "Michael Scarn");
+    values.put("Today, smoking is going to ", "save lives");
+    values.put("Ryan started ", "the fire");
+    values.put("Fool me once, strike one. Fool me twice, ", "strike three");
+    values.put("Well, well, well, ", "how the turntables");
+  }
+
+  public String getCurrent() {
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java:59]: getCurrentPrompt

File: portfolio/src/main/webapp/theoffice/theoffice.js [hunk 4]
```diff
@@ -13,33 +13,26 @@
...
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+function getOfficeGame() {/* exported getOfficeGame */
+  fetch('/office-game').then((response) => response.json()).then((game) => {
+    const state = game.state;
+    console.log(state);
```
Comment (chen-dawn) [inline portfolio/src/main/webapp/theoffice/theoffice.js:21]: remove console.log statements

File: portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+  /**
+   * "Submits" the answer by checking if it is equal to the answer and assigning
+   * that to lastSubmission variable
+   */
+  public void submitAnswer(String guess) {
+    lastSubmission = values.get(current).equals(guess);
+  }
+
+  public boolean getSubmission() {
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java:75]: This naming is kinda confusing. Maybe something like `getWinStatus`?

File: portfolio/src/main/java/com/google/sps/servlets/OfficeServlet.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+  private int state = 0;
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    // state 0 signifies they submit their answer, state 1 signifies saying whether they won
+    if (state == 0) {
+      game.newChoice();
+      String json = "{ \"state\" : " + state + ",  \"quote\" : \"" + game.getCurrent() + "\" } ";
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/OfficeServlet.java:38]: Prefer not to use a string representation - either use a JSON object or define a private `GameState` class in this OfficeServlet.

File: portfolio/src/main/java/com/google/sps/servlets/OfficeServlet.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+/** Servlet that encapsulates the Office game */
+@WebServlet("/office-game")
+public final class OfficeServlet extends HttpServlet {
+  private OfficeGame game = new OfficeGame();
+
+  private int state = 0;
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/OfficeServlet.java:33]: What does this function do? Add top level javadoc.

File: portfolio/src/main/java/com/google/sps/servlets/OfficeServlet.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+      String json = "{ \"state\" : " + state + ", \"correct\" : " + game.getSubmission()
+          + ", \"answer\" : \"" + game.getAnswer() + "\"}";
+      response.getWriter().println(json);
+      state = 0;
+    }
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // they submit their answer and the game switches to state 1
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/OfficeServlet.java:50]: Add function level description on what this function does. For example
"Gets user answer and submits it to the game. Then redirect users to the results page."

File: portfolio/src/main/webapp/theoffice/theoffice.css [hunk 9]
```diff
@@ -1,3 +1,22 @@
...
+  margin: 4px 2px;
+  padding: 16px 32px;
+  text-align: center;
+  text-decoration: none;
+  transition-duration: 0.4s;
+}
+
+button:hover {
+  background-color: #800080;
+  color: #ffffff;
```
Comment (chen-dawn) [inline portfolio/src/main/webapp/theoffice/theoffice.css:17]: prefer 3 character hexcodes. Here and below. https://google.github.io/styleguide/htmlcssguide.html#Hexadecimal_Notation

File: portfolio/src/main/webapp/theoffice/theoffice.html [hunk 10]
```diff
@@ -7,16 +7,22 @@
...
-      <h1>Click here to get a random Office quote:</h1>
-      <a class="button button-purple" onclick="addRandomGreeting()">Click me!</a>
-      <!-- [html-validate-disable empty-heading] -->
-      <h3 id="greeting-container"></h3>
-      <!-- [html-validate-enable empty-heading] -->
+      <div id="game-box">
+        <br /><br />
+        <h3 id="title">Finish the quote!</h3>
+        <p id="quote"></p>
+        <!-- [html-validate-disable attribute-allowed-values] -->
```
Comment (chen-dawn) [inline portfolio/src/main/webapp/theoffice/theoffice.html:19]: What's throwing this error?

File: portfolio/src/main/webapp/theoffice/theoffice.js [hunk 11]
```diff
@@ -13,33 +13,26 @@
...
+    // state 0 signifies submission
+    // state 1 signifies saying whether they got it right
+    if (state == 0) {
```
Comment (chen-dawn) [inline portfolio/src/main/webapp/theoffice/theoffice.js:24]: Same here, not super clear what state = 0 and what state = 1 means. Perhaps use the same enum definitions.

File: portfolio/src/main/webapp/theoffice/theoffice.js [hunk 12]
```diff
@@ -13,33 +13,26 @@
...
+      const paragraph = document.getElementById('quote');
```
Comment (chen-dawn) [inline portfolio/src/main/webapp/theoffice/theoffice.js:25]: more descriptive names, e.g. `quoteContainer`. `paragraph` is ambiguous

**Pull request merged**